lib/internal/test_runner/reporter/spec.js

```diff
@@ -103,6 +106,17 @@ class SpecReporter extends Transform {
         break;
       case 'test:diagnostic':
         return `${color}${this.#indent(data.nesting)}${symbol}${data.message}${white}\n`;
+      case 'test:eot':
```

Comment:
I think we can perform this logic inside [`_flush`](https://nodejs.org/api/stream.html#transform_flushcallback)
this way no need to introduce a new `eot` event
also, if we extract the logic of reporting a test to a method we can re-use it and receive a more consistent output:
https://github.com/nodejs/node/blob/main/lib/internal/test_runner/reporter/spec.js#L85-L99

Reply:
After thinking for a while,

I think that we can specifically extract the formatting logic because it is what we want to re-use.
But it seems cleaner if we keep the [handling](https://github.com/nodejs/node/blob/843856ead771a2db21d63f65eb1d52ddb264d9dd/lib/internal/test_runner/reporter/spec.js#L70-L84) of recursive test in the original place, including [89~93](https://github.com/nodejs/node/blob/843856ead771a2db21d63f65eb1d52ddb264d9dd/lib/internal/test_runner/reporter/spec.js#L89-L93).